src/symbolize/symbolizer.rs

```diff
@@ -879,6 +917,75 @@ impl Symbolizer {
         }
     }
 
+    fn create_vdso_resolver(&self, pid: Pid, range: &Range<Addr>) -> Result<ElfResolver> {
+        let start = range.start;
+        let size = range.end.saturating_sub(start);
+        // Put in some upper limit on the vDSO data that we are going to
+        // store directly. In most cases this should be a single 4 KiB
+        // page, maybe two. Assume a worst case of one 4 MiB page, which
+        // should provide plenty of slack.
+        if size > 4 * 1024 * 1024 {
+            return Err(Error::with_invalid_data(format!(
+                "vDSO of process {pid} is unexpectedly large ({size} bytes)"
+            )))
+        }
+
+        // Read the vDSO image.
+        let path = format!("/proc/{pid}/mem");
+        let mut mem =
+            File::open(&path).with_context(|| format!("failed to open `{path}` for reading"))?;
+        let _prev = mem
+            .seek(SeekFrom::Start(start))
+            .with_context(|| format!("failed to seek `{path}`"))?;
+
+        let mut vdso_buf = Vec::new();
+        let () = vdso_buf.resize(size as _, 0u8);
```

Comment:
Just curious because I have never seen this before, what's the purpose of the empty tuple?

Reply:
I use it to make it obvious that the function does not return anything (that we would otherwise silently discard).